### DISCUSSION TITLE
Does Livewire components support only single event listener on component tag element?

### DISCUSSION THREAD

**Discussion by stescacom** [livewire/livewire#8097] [1. Help]
I was trying to listen to multiple events on the component element tag.

```
<livewire:form-component @close="..." @created="..." />

// inside component

public function process()
{
    $this->dispatch('created');
}

<button @click="$dispatch('close')">Close</button>
<button wire:click="process">Save</button>

```
When it is just `@close` the method is called. But when I add `@created` nothing happens. Why is that? Is it limited to single event listeneing?

**Comment by danie-ramdhani** [comment]
why not using livewire attributes?

```php
#[\Livewire\Attributes\On('created')]
public function created()
{
    //
}

#[\Livewire\Attributes\On('close')]
public function close()
{
    //
}
```

**Reply by stescacom** [reply]
I did, and that worked.. but I am curious about tag event listening.

**Reply by danie-ramdhani** [reply]
if you inspect the element, livewire component only has `x-on:created="..."`. idk maybe it is a bug...

**Comment by joshhanley** [comment]
@stescacom I'm not sure if it would work or not, I suspect that it probably doesn't. But could be useful PRs welcome! 🙂

**Reply by neeravp** [reply]
I can confirm it doesn't work. Using `#[On]` attribute is the only solution. However it does encapsulate things and makes it hard when revisiting the code after a few months. 

I have started writing comments in the blade views for reference. 

I would request you to look and see if it is possible to have multiple events on the component itself to make life easier. Thanks

Great work you all maintainers are doing

cc: @joshhanley @calebporzio

**Reply by joshhanley** [reply]
@neeravp this got added by @danie-ramdhani in v3.4.12, so please update and test again 🙂 PR for reference PR #8242

**Reply by neeravp** [reply]
Thanks for the update. I was on v3.4.9. Have upgraded to v3.5.4 and it works now. Thanks a lot.
